.github/codecov.yml

```diff
@@ -0,0 +1,52 @@
+comment:
+  layout: "header, diff, components, files" # show component info in the PR comment
+component_management:
+  default_rules: # default rules that will be inherited by all components
+    statuses:
+      - type: project # in this case every component that doens't have a status defined will have a project type
+        target: auto
+        threshold: 5%
+        informational: true
+        branches:
+          - master
+  individual_components:
+    - component_id: backend-apis
+      name: backend-apis # only Golang backend APIs
+      paths:
+        - "webv2/**"
+        - "!webv2/webCmd.go"
+      statuses: # each component has its own status and corresponding configuration
+        - type: project
+          target: auto
+          informational: true
+    - component_id: backend-library
+      name: backend-library # only library code
+      paths:
+        - "!webv2/**"
+        - "!ui/**"
+        - "!cmd/**"
+      statuses: # each component has its own status and corresponding configuration
+        - type: project
+          target: auto
+          informational: true
+    - component_id: cli 
+      name: cli # only the CLI
+      paths:
+        - "cmd/**"
+        - "webv2/webCmd.go"
+      statuses: # each component has its own status and corresponding configuration
+        - type: project
+          target: auto
+          informational: true
+    - component_id: frontend
+      name: frontend # only frontend components
+      paths:
+        - ui/**
+      statuses: # the core component has its own statuses
+        - type: project
+          target: auto
+          informational: true
```

Comment:
Frontend end tests will refer to a report different from backend end tests , it might make sense to use flags: https://docs.codecov.com/docs/flags instead of components

Reply:
As per https://docs.codecov.com/docs/components I looks me to that the `codecov` uploader will automatically find relevant coverage reports and send them to `CodeCov` backend.
After that, the virtual filters in the `codecav.yml` would be applicable for all the collected reports.